### ISSUE TITLE
error when run helloapache

### ISSUE THREAD

**Issue by tkdchen** [projectatomic/atomicapp#121]
I use vagrant to build a clean machine to try installation of helloapache. Unfortunately, ran into this error.

OS: CentOS 7.1
Both `atomic` and `kubernetes` are installed and all `kubernetes` services start

```
[vagrant@localhost ~]$ sudo atomic run projectatomic/helloapache                                                                                       [85/675]
latest: Pulling from docker.io/projectatomic/helloapache
6941bfcbbfca: Pull complete 
41459f052977: Pull complete 
fd44297e2ddb: Pull complete 
20d1e586344c: Pull complete 
3a1c52000255: Pull complete 
5f28bf0b404e: Pull complete 
96ff8def7982: Pull complete 
3ed5773a0092: Pull complete 
23c83502f560: Pull complete 
8d47527a4940: Pull complete 
885a78fe8082: Pull complete 
017d25d95b71: Pull complete 
f04bc86c4c76: Pull complete 
2e9e9f97edbf: Pull complete 
77c9a71db428: Pull complete 
fe5b4f2b041b: Pull complete 
2319dcd5fbc9: Pull complete 
67841bf3065c: Pull complete 
c91abd5ddb44: Pull complete 
dfd781052769: Already exists 
Digest: sha256:be520aeff99cd672851b96e78daa93b97169f471fe8fefd00d957212d2faffc8
Status: Downloaded newer image for docker.io/projectatomic/helloapache:latest
docker run -it --rm --privileged --net=host -v `pwd`:/atomicapp -v /run:/run -v /:/host --name helloapache -e NAME=helloapache -e IMAGE=projectatomic/helloapac
he projectatomic/helloapache -v run /atomicapp
2015-05-25 10:12:13,525 - atomicapp.install - INFO - App name is projectatomic/helloapache, will be populated to /atomicapp
2015-05-25 10:12:13,525 - atomicapp.utils - DEBUG - {'provider': 'kubernetes'}
2015-05-25 10:12:13,570 - atomicapp.utils - DEBUG - Output of docker images cmd: 
latest: Pulling from docker.io/projectatomic/helloapache
6941bfcbbfca: Already exists 
41459f052977: Already exists 
fd44297e2ddb: Already exists 
20d1e586344c: Already exists 
3a1c52000255: Already exists 
5f28bf0b404e: Already exists 
96ff8def7982: Already exists 
3ed5773a0092: Already exists
23c83502f560: Already exists                                                                                                                           [47/675]
8d47527a4940: Already exists 
885a78fe8082: Already exists 
017d25d95b71: Already exists 
f04bc86c4c76: Already exists 
2e9e9f97edbf: Already exists 
77c9a71db428: Already exists 
fe5b4f2b041b: Already exists 
2319dcd5fbc9: Already exists 
67841bf3065c: Already exists 
c91abd5ddb44: Already exists 
dfd781052769: Already exists 
Digest: sha256:be520aeff99cd672851b96e78daa93b97169f471fe8fefd00d957212d2faffc8
Status: Image is up to date for docker.io/projectatomic/helloapache:latest
2015-05-25 10:12:23,786 - atomicapp.utils - DEBUG - {'provider': 'kubernetes'}
2015-05-25 10:12:23,787 - atomicapp.install - DEBUG - Creating a container with name helloapache-XkgFjD
131266b238d10f53ed2f74ed5976109b7c5d429d3da1eabebfd6061f7ae54fa5
2015-05-25 10:12:24,001 - atomicapp.utils - INFO - Using temporary directory /tmp/appent-helloapacheOBrG1k
2015-05-25 10:12:24,002 - atomicapp.install - DEBUG - ['docker', 'cp', 'helloapache-XkgFjD:/application-entity', '/tmp/appent-helloapacheOBrG1k']
2015-05-25 10:12:24,060 - atomicapp.install - DEBUG - Application entity data copied to /tmp/appent-helloapacheOBrG1k
helloapache-XkgFjD
2015-05-25 10:12:24,948 - atomicapp.install - DEBUG - nulecule path for pulled image: /tmp/appent-helloapacheOBrG1k/application-entity/nulecule
2015-05-25 10:12:24,953 - atomicapp.params - DEBUG - Setting app id to helloapache-app
2015-05-25 10:12:24,953 - atomicapp.install - DEBUG - App ID: helloapache-app
2015-05-25 10:12:24,953 - atomicapp.install - INFO - Copying app helloapache
2015-05-25 10:12:24,956 - atomicapp.utils - DEBUG - Provider: kubernetes
2015-05-25 10:12:24,956 - atomicapp.utils - DEBUG - Artifact file://artifacts/k8s/hello-apache-pod.json: OK
2015-05-25 10:12:24,956 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
2015-05-25 10:12:24,956 - atomicapp.install - INFO - Installing dependencies for helloapache-app
2015-05-25 10:12:24,956 - atomicapp.utils - DEBUG - {u'artifacts': {u'kubernetes': [u'file://artifacts/k8s/hello-apache-pod.json']}, u'params': {u'image': {u'd
efault': u'centos/httpd', u'description': u'The webserver image'}, u'hostport': {u'default': 80, u'description': u'The host TCP port as the external endpoint'}
}}
2015-05-25 10:12:24,956 - atomicapp.install - DEBUG - Component helloapache-app is part of the app
2015-05-25 10:12:24,957 - atomicapp.install - DEBUG - Values: {u'helloapache-app': {u'image': u'centos/httpd', u'hostport': 80}}
2015-05-25 10:12:24,957 - atomicapp.install - DEBUG - {u'helloapache-app': {u'image': u'centos/httpd', u'hostport': 80}}
2015-05-25 10:12:24,957 - atomicapp.params - DEBUG - Data given {u'helloapache-app': {u'image': u'centos/httpd', u'hostport': 80}}
2015-05-25 10:12:24,957 - atomicapp.install - DEBUG - {u'helloapache-app': {u'image': u'centos/httpd', u'hostport': 80}, 'general': {'provider': 'kubernetes'}}
2015-05-25 10:12:24,957 - atomicapp.params - INFO - Writing answers file template to /atomicapp/answers.conf.sample
2015-05-25 10:12:24,957 - atomicapp.utils - DEBUG - /atomicapp/.workdir                                                                                 [9/675]
2015-05-25 10:12:24,957 - atomicapp.plugin - DEBUG - Loading providers from /usr/lib/python2.7/site-packages/atomicapp/providers
2015-05-25 10:12:24,964 - atomicapp.params - DEBUG - Setting app id to helloapache-app
2015-05-25 10:12:24,964 - atomicapp.utils - INFO - Version check successful: specversion == 0.0.1-alpha
2015-05-25 10:12:24,964 - atomicapp.utils - DEBUG - Provider: kubernetes
2015-05-25 10:12:24,964 - atomicapp.utils - DEBUG - Artifact file://artifacts/k8s/hello-apache-pod.json: OK
2015-05-25 10:12:24,964 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
2015-05-25 10:12:24,964 - atomicapp.utils - DEBUG - {u'artifacts': {u'kubernetes': [u'file://artifacts/k8s/hello-apache-pod.json']}, u'params': {u'image': {u'd
efault': u'centos/httpd', u'description': u'The webserver image'}, u'hostport': {u'default': 80, u'description': u'The host TCP port as the external endpoint'}
}}
2015-05-25 10:12:24,964 - atomicapp.run - DEBUG - Processing component helloapache-app
2015-05-25 10:12:24,964 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-05-25 10:12:24,964 - atomicapp.params - DEBUG - Param provider already in general with value kubernetes
2015-05-25 10:12:24,964 - atomicapp.run - INFO - Using provider kubernetes for component helloapache-app
2015-05-25 10:12:24,965 - atomicapp.run - DEBUG - Templating artifact /atomicapp/artifacts/k8s/hello-apache-pod.json
2015-05-25 10:12:24,965 - atomicapp.params - DEBUG - Param provider already in general with value kubernetes
2015-05-25 10:12:24,965 - atomicapp.run - DEBUG - Config: {u'image': u'centos/httpd', 'provider': 'kubernetes', u'hostport': 80} 
2015-05-25 10:12:24,965 - atomicapp.run - DEBUG - {u'image': u'centos/httpd', 'provider': 'kubernetes', u'hostport': 80}
2015-05-25 10:12:24,965 - atomicapp.plugin - DEBUG - Writing artifact to /atomicapp/.workdir/helloapache-app/artifacts/k8s/hello-apache-pod.json
2015-05-25 10:12:24,966 - kubernetes - DEBUG - /atomicapp/.workdir/helloapache-app/artifacts/k8s/hello-apache-pod.json
pods/helloapache
Error: pods "helloapache" already exists
Traceback (most recent call last):
  File "/usr/bin/atomicapp", line 9, in <module>
    load_entry_point('atomicapp==0.1', 'console_scripts', 'atomicapp')()
  File "/usr/lib/python2.7/site-packages/atomicapp/cli/main.py", line 111, in main
    cli.run()
  File "/usr/lib/python2.7/site-packages/atomicapp/cli/main.py", line 92, in run
    args.func(args)
  File "/usr/lib/python2.7/site-packages/atomicapp/cli/main.py", line 33, in cli_run
    ae.run()
  File "/usr/lib/python2.7/site-packages/atomicapp/run.py", line 180, in run
    self._dispatchGraph()
  File "/usr/lib/python2.7/site-packages/atomicapp/run.py", line 95, in _dispatchGraph
    self._processComponent(component, graph_item)
  File "/usr/lib/python2.7/site-packages/atomicapp/run.py", line 165, in _processComponent
    provider.deploy()
  File "/usr/lib/python2.7/site-packages/atomicapp/providers/kubernetes.py", line 73, in deploy
    self._callK8s(k8s_file)
  File "/usr/lib/python2.7/site-packages/atomicapp/providers/kubernetes.py", line 35, in _callK8s
    subprocess.check_call(cmd)
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/host/usr/bin/kubectl', 'create', '-f', u'/atomicapp/.workdir/helloapache-app/artifacts/k8s/hello-apache-pod.json']' 
returned non-zero exit status 1
```

**Comment by goern** [comment]
pls check that the helloapache pod does not exists (maybe due to a previous run?). `kubectl get pods` and `kubectl get services` is your friend.

**Comment by vpavlin** [comment]
This will be fixed by #124 - kubectl command is run twice by provider

**Comment by goern** [comment]
merged #124 and fixed #121
